text/0000-unnamed-fields.md

```diff
@@ -0,0 +1,463 @@
+name, allowing disambiguation via a single token of context (`union {`).
+
+## Layout and Alignment
+
+The layout and alignment of a `struct` or `union` containing unnamed
+fields should look the same as if each unnamed field has a separately declared
+type and a named field of that type, rather than as if the fields appeared
+directly within the containing `struct` or `union`. In some cases, this may
+result in different alignment.
+
+## Simultaneous Borrows
+
+An unnamed `struct` within a `union` should behave the same with respect to
+borrows as a named and typed `struct` within a `union`, allowing borrows of
+multiple fields from within the `struct`, while not permitting borrows of other
+fields in the `union`.
+
+## Visibility
+
+Each field within an unnamed `struct` or `union` may have an attached
+visibility (`pub` or `pub(crate)`). An unnamed field itself does not have its
+own visibility; all of its fields appear directly within the containing
+structure, and their own visibilities apply.
+
+## Documentation
+
+Public fields within an unnamed `struct` or `union` should appear in the
+rustdoc documentation of the outer structure, along with any doc comment or
+attribute attached to those fields. The rendering should include all unnamed
+fields that contain (at any level of nesting) a public field, and should
+include the `// some fields omitted` note within any `struct` or `union` that
+has non-public fields, including unnamed fields.
+
+Any unnamed field that contains only non-public fields should be omitted
+entirely, rather than included with its fields omitted. Omitting an unnamed
+field should trigger the `// some fields omitted` note.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This introduces additional complexity in structure definitions. Strictly
+speaking, C interfaces do not *require* this mechanism; any such interface
+*could* define named struct or union types, and define named fields of that
+type. This RFC provides a usability improvement for such interfaces.
+
+# Rationale and Alternatives
+[alternatives]: #alternatives
+
+## Not implementing this feature at all
+
+Choosing not to implement this feature would force binding generators (and the
+authors of manual bindings) to invent new names for these groupings of fields.
+Users would need to look up the names for those groupings, and would not be
+able to rely on documentation for the underlying interface. Furthermore,
+binding generators would not have any basis on which to generate a meaningful
+name.
+
+## Not implementable as a macro
+
+We cannot implement this feature as a macro, because it affects the names used
+to reference the fields contained within an unnamed field. A macro could
+extract and define types for the unnamed fields, but that macro would have to
+give a name to those unnamed fields, and accesses would have to include the
+intermediate name.
+
+## Unnamed fields using `_` with named types
```

Comment:
@est31 This would *exclusively* be syntax for declaring structs and unions. If some future general mechanism for anonymous types came along, it'd look more like this:

```
fn f(value: struct { x: u32, y: u32 }) {
    println!("{} {}", value.x, value.y);
}

fn main() {
    f({ x: 5, y: 10 });
}
```